cmis_write/ir_attachment.py

```diff
@@ -0,0 +1,243 @@
+
+    def create(self, cr, uid, values, context=None):
+        metadata_obj = self.pool.get('metadata')
+        user_obj = self.pool.get('res.users')
+        user_login = user_obj.browse(cr, uid, uid, context=context).login
+        session = ConnectorSession(cr, uid, context=context)
+        value = {
+            'name': values.get('name'),
+            'datas_fname': values.get('datas_fname'),
+            'file_type': values.get('file_type') or '',
+            'datas': values.get('datas'),
+            'description': values.get('description') or '',
+
+        }
+
+        metadata_ids = metadata_obj.search(cr, uid, [], context=context)
+        dict_metadata = {}
+        list_fields = []
+        # Get list of metadata
+        if values.get('res_model'):
+            for line in metadata_obj.browse(cr, uid, metadata_ids,
+                                            context=context):
+                if line.model_id.model == values.get('res_model'):
+                    if line.metadata_list_ids:
+                        for one_field in line.metadata_list_ids:
+                            list_fields.append(one_field.field_id.name)
+            result = self.pool.get(values.get('res_model')).read(cr, uid, [
+                values.get('res_id')], list_fields, context=context)[0]
+
+            for one_field in list_fields:
+                dict_metadata['cmis:' + one_field] = result[one_field]
+        values['datas'] = None
+        res = super(ir_attachment, self).create(cr, uid, values,
+                                                context=context)
+        # Create Job
+        # if bool_testdoc in context, we don't need to create
+        # the doc in the DMS
+        if not context.get('bool_testdoc'):
+            create_doc_in_edm.delay(
+                session, 'ir.attachment', value, res, dict_metadata,
+                user_login)
+        return res
+
+    def action_download(self, cr, uid, ids, context=None):
+        """
+        Download document if it is available in one of backend
+        """
+        if context is None:
+            context = {}
+        cmis_backend_obj = self.pool.get('cmis.backend')
+        # login with the cmis account
+        backend_ids = cmis_backend_obj.search(cr, uid,
+                                              [('storing_ok', '=', 'True')],
+                                              context=context)
+        for backend in cmis_backend_obj.browse(cr, uid, backend_ids,
+                                               context=context):
+            try:
+                repo = cmis_backend_obj._auth(cr, uid, backend.id,
+                                              context=context)
+                cmis_backend_rec = self.read(
+                    cr, uid, ids, ['id_dms'], context=context)[0]
+                id_dms = cmis_backend_rec['id_dms']
+                # Get results from id of document
+                results = repo.query(" SELECT * FROM  cmis:document WHERE \
```

Comment:
I have tested this, and retrieves an empty list. What works instead of the query is:
result = repo.getObject(id_dms)